build_tools/fetch_artifacts.py

```diff
@@ -64,11 +96,16 @@ def retrieve_base_artifacts(args, run_id, build_dir):
     if args.blas:
         base_artifacts.append("host-blas_lib")
 
-    for base_artifact in base_artifacts:
-        s3_exec(GENERIC_VARIANT, base_artifact, run_id, build_dir)
+    cmds = s3_commands_for_artifacts(base_artifacts, run_id, build_dir, GENERIC_VARIANT)
+    parallel_exec_commands(cmds)
 
 
 def retrieve_enabled_artifacts(args, target, run_id, build_dir):
+    """
+    This function retrieves TheRock artifacts using AWS S3 copy, based on the enabled arguments from `args`
+    If no artifacts have been collected, we assume that we want to install all artifacts
```

Comment:
newline between docstring summary and description

https://peps.python.org/pep-0257/#multi-line-docstrings

> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description.